org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/GradleProjectImporter.java

```diff
@@ -186,14 +213,32 @@ public void importToWorkspace(IProgressMonitor monitor) throws CoreException {
 				JavaLanguageServerPlugin.logException("Failed to update digest for gradle build file", e);
 			}
 		});
+		for (IProject gradleProject : ProjectUtils.getGradleProjects()) {
+			gradleProject.deleteMarkers(COMPATIBILITY_MARKER_ID, true, IResource.DEPTH_INFINITE);
+		}
+		if (compatibilityStatus.getChildren().length > 0) {
```

Comment:
No need to check the length of the children beforehand. for-loop will cover it as well.

Reply:
removed in https://github.com/eclipse/eclipse.jdt.ls/pull/1965/commits/87298b495b093411006826cc69f1775704fea69d